SimDataFormats/Track/src/classes_def.xml

```diff
@@ -4,22 +4,24 @@
   <class name="CoreSimTrack" ClassVersion="10">
    <version ClassVersion="10" checksum="3936841839"/>
   </class>
-  <class name="SimTrack" ClassVersion="13">
+  <class name="SimTrack" ClassVersion="14">
+   <version ClassVersion="14" checksum="2517401874"/>
    <version ClassVersion="13" checksum="1912247222"/>
    <version ClassVersion="12" checksum="3470347245"/>
    <version ClassVersion="11" checksum="1785575744"/>
    <version ClassVersion="10" checksum="1430205451"/>
    <ioread sourceClass = "SimTrack" version="[-12]" targetClass="SimTrack" source="" target="">
     <![CDATA[
-        SimTrack tmp(newObj->type(), newObj->momentum(), newObj->vertIndex(), newObj->genpartIndex(), newObj->trackerSurfacePosition(), newObj->trackerSurfaceMomentum());
+        SimTrack tmp(newObj->type(), newObj->momentum(), newObj->vertIndex(), newObj->getPrimaryID(), newObj->trackerSurfacePosition(), newObj->trackerSurfaceMomentum());
         tmp.setTrackId(newObj->trackId());
         tmp.setEventId(newObj->eventId());
-        tmp.setCrossedBoundaryVars(
-            newObj->crossedBoundary(), newObj->getIDAtBoundary(), newObj->getPositionAtBoundary(), newObj->getMomentumAtBoundary());
-        if (newObj->isFromBackScattering()) {
-          tmp.setFromBackScattering();
-        }
-        if (newObj->genpartIndex() != -1) {
+        if (newObj->crossedBoundaryOld())
+          tmp.setCrossedBoundaryVars(
+            true, newObj->getIDAtBoundary(), newObj->getPositionAtBoundary(), newObj->getMomentumAtBoundary());
+        else
+          tmp.setCrossedBoundaryVars(
+            false, newObj->getIDAtBoundary(), newObj->getPositionAtBoundary(), newObj->getMomentumAtBoundary());
```

Comment:
it is possible to access the member variables from the old version of the object. in this case, you should be able to do this:
```
tmp.setCrossedBoundaryVars(
            onfile.crossedBoundary_, newObj->getIDAtBoundary(), newObj->getPositionAtBoundary(), newObj->getMomentumAtBoundary());
```
then you can avoid restoring the old member variable in the current object definition.